node-manager/src/dlc.rs

```diff
@@ -0,0 +1,275 @@
+use crate::localstorage::MutinyBrowserStorage;
+use bdk::blockchain::EsploraBlockchain;
+use bdk::database::Database;
+use bdk::keys::ExtendedKey;
+use bdk::wallet::AddressIndex;
+use bdk::{SignOptions, Wallet};
+use bdk_macros::maybe_await;
+use bip39::Mnemonic;
+use bitcoin::psbt::PartiallySignedTransaction;
+use bitcoin::util::address::Address;
+use bitcoin::util::bip32::{ChildNumber, DerivationPath, ExtendedPrivKey};
+use bitcoin::{Block, Network, Script, Transaction, TxIn, Witness};
+use dlc_manager::error::Error as ManagerError;
+use dlc_manager::error::Error::WalletError;
+use dlc_manager::{Blockchain, Utxo};
+use log::error;
+use secp256k1::Secp256k1;
+use std::str::FromStr;
+use std::sync::Arc;
+use wasm_bindgen_futures::spawn_local;
+
+#[derive(Debug)]
+pub struct MutinyDLCWallet {
+    wallet: Arc<Wallet<MutinyBrowserStorage>>,
+    blockchain: Arc<EsploraBlockchain>,
+    storage: Arc<MutinyBrowserStorage>,
+    mnemonic: Mnemonic,
+}
+
+fn bdk_err_to_manager_err(e: bdk::Error) -> ManagerError {
+    WalletError(Box::new(e))
+}
+
+impl MutinyDLCWallet {
+    fn derive_secret_key(&self, index: u32) -> secp256k1::SecretKey {
+        let entropy = self.mnemonic.to_entropy();
+        let network = self.wallet.network();
+        let xprivkey = ExtendedPrivKey::new_master(network, &entropy).unwrap();
+        let xkey: ExtendedKey = ExtendedKey::from(xprivkey);
+        let master_xprv = xkey.into_xprv(network).unwrap();
+        let coin_type = match network {
+            Network::Bitcoin => 0,
+            Network::Testnet => 1,
+            Network::Signet => 1,
+            Network::Regtest => 1,
+        };
```

Comment:
Weird that you had to do this, I put some of this in `utils` though if you want to move it there though. No worries if not.